prow/tide/tide_test.go

```diff
@@ -80,22 +80,31 @@ func TestAccumulateBatch(t *testing.T) {
 		},
 		{
 			name:       "batch pending",
-			presubmits: map[int]sets.String{1: jobSet, 2: jobSet},
+			presubmits: map[int]sets.String{1: sets.NewString("foo"), 2: sets.NewString("foo")},
 			pulls:      []pull{{1, "a"}, {2, "b"}},
 			prowJobs:   []prowjob{{job: "foo", state: kube.PendingState, prs: []pull{{1, "a"}}}},
 			pending:    true,
 		},
+		{
+			name:       "pending batch missing presubmits is ignored",
+			presubmits: map[int]sets.String{1: jobSet},
+			pulls:      []pull{{1, "a"}, {2, "b"}},
+			prowJobs:   []prowjob{{job: "foo", state: kube.PendingState, prs: []pull{{1, "a"}}}},
+		},
 		{
 			name:       "batch pending, successful previous run",
```

Comment:
I think I'm misunderstanding this test - if PR 2 succeeded but we have a pending batch, shouldn't we still wait, not merge 2?

Reply:
What is the benefit of waiting for a pending batch to complete if we already have a successful batch that we can merge? Once we add support for triggering multiple batches of different sizes we'll want to update this to consider the size of the batches, but for now it seems appropriate to just merge the successful batch. Note that we are not triggering multiple batches currently so this case shouldn't really occur anyways.

Reply:
Oh, as there was only one PR, I thought there was one batch and one singleton test. As they're both batches, that makes sense